File: NDPCommon/src/main/java/com/slender/validator/PhoneValidator.java
package com.slender.validator;
import com.slender.annotation.Phone;
import jakarta.validation.ConstraintValidator;
import jakarta.validation.ConstraintValidatorContext;

import java.util.regex.Pattern;

public class PhoneValidator implements ConstraintValidator<Phone, String> {
    @Override
    public boolean isValid(String phone, ConstraintValidatorContext context) {
        if (phone == null || phone.isEmpty()) return true;
        return Pattern.compile("^1[3-9]\\d{9}$").matcher(phone).matches();
    }
}

File: NDPCommon/src/main/java/com/slender/validator/PhoneValidator.kt
package com.slender.validator

import jakarta.validation.ConstraintValidator
import com.slender.annotation.Phone
import jakarta.validation.ConstraintValidatorContext
import java.util.regex.Pattern

class PhoneValidator : ConstraintValidator<Phone, String> {
    override fun isValid(phone: String?, context: ConstraintValidatorContext?): Boolean {
        if (phone == null) return true
        return phone.matches(Regex("^1[3-9]\\d{9}$"))
    }
}
